src/main/java/spoon/support/visitor/java/JavaReflectionVisitorImpl.java

```diff
@@ -61,7 +61,15 @@ public void visitPackage(Package aPackage) {
 			visitField(field);
 		}
 		for (Class<?> aClass : clazz.getDeclaredClasses()) {
-			visitClass(aClass);
+			if (aClass.isAnnotation()) {
+				visitAnnotationClass((Class<Annotation>) aClass);
```

Comment:
I think it will produce class cast exception.
The correct type is `Class<? extends Annotation>`

Reply:
I get this code from there: https://github.com/INRIA/spoon/blob/cef5e079f335f48133395cb1f369ce1e0cdf83cc/src/main/java/spoon/support/visitor/java/JavaReflectionTreeBuilder.java#L88

and apparently it's used for 2 years... But PR with a failing test case is welcome :)